chess/src/history.rs

```diff
@@ -18,7 +19,21 @@ impl History {
 
     pub fn add_info(&mut self, next_state: BoardState, mv: Move) {
         self.turns.push(Turn::new(next_state, mv));
-        self.current_turn += 1
+        self.current_turn += 1;
+        let is_pawn = next_state.get_piece(&mv.to).unwrap().is_pawn();
```

Comment:
So assuming promoting a pawn counts as a pawn move, the `mv.to` square would no longer be a pawn in that case. So maybe we should check if `mv.from` in the previous state was a pawn instead.

Reply:
big brain i oversaw that. imma test this out ! thx dan